src/fsharp/ConstraintSolver.fs

```diff
@@ -1446,7 +1446,10 @@ and GetRelevantMethodsForTrait (csenv:ConstraintSolverEnv) permitWeakResolution
             /// to a generic instantiation for an operator based on the right hand type. 
             
             let minfos = List.reduce (ListSet.unionFavourLeft MethInfo.MethInfosUseIdenticalDefinitions) minfos
+            let relevantMinfos = minfos |> List.filter(fun minfo -> not minfo.IsDispatchSlot && not minfo.IsVirtual && minfo.IsInstance)
             minfos
+            |> List.filter(fun minfo1 ->
+                not(relevantMinfos |> List.exists (fun minfo2 -> minfo1.IsDispatchSlot && MethInfosEquivByNameAndSig EraseAll true csenv.g csenv.amap m minfo2 minfo1)))
```

Comment:
Please test `minfo1.IsDispatchSlot` before doing the `List.exists` call since it is independent of the predicate

Reply:
Please also add some comments here :)

Reply:
ok thanks will do